src/memory/virtualMemory.ts

```diff
@@ -21,6 +17,7 @@ export class VirtualMemory {
   >();
 
   private rbp: number;
+  private readonly rdx: number;
```

Comment:
The `rdx` register is typically used to pass the 3rd argument to functions in the 64-bit x86 architecture. It doesn't look like it uses the same semantics here.

In fact, if it can be made `readonly`, it sounds like it should be some fixed constant rather than a register. Accordingly, we shouldn't name it like a register.